remotes/docker/scope.go

```diff
@@ -45,32 +46,84 @@ func repositoryScope(refspec reference.Spec, push bool) (string, error) {
 type tokenScopesKey struct{}
 
 // contextWithRepositoryScope returns a context with tokenScopesKey{} and the repository scope value.
-func contextWithRepositoryScope(ctx context.Context, refspec reference.Spec, push bool) (context.Context, error) {
+func contextWithRepositoryScope(ctx context.Context, refspec reference.Spec, push bool, combineWithExistingScopes bool) (context.Context, error) {
 	s, err := repositoryScope(refspec, push)
 	if err != nil {
 		return nil, err
 	}
-	return context.WithValue(ctx, tokenScopesKey{}, []string{s}), nil
+	scopes := []string{s}
+	if combineWithExistingScopes {
```

Comment:
Would there be a downside to always doing this? It would mean simpler code here and in callers (no need for the new arg) and is what I think I would always expect of a function which takes a `context` and produces another one -- i.e. that they are cumulative.